### ISSUE TITLE
Can this module tunnel RTSP over HTTP?

### ISSUE THREAD

**Issue by jandrieu** [iizukanao/node-rtsp-rtmp-server#21]
The documentation doesn't have an example for taking an incoming RTSP stream and publishing it via HTTP. Looking at the code it seems like http tunneling is supported for posting an RTSP stream... but I didn't see how I would tunnel RTSP over HTTP as per the QuickTime spec: http://www.opensource.apple.com/source/QuickTimeStreamingServer/QuickTimeStreamingServer-412.42/Documentation/RTSP_Over_HTTP.pdf 

Any guidance? Is this already in this module or am I looking for a feature that isn't there?

**Comment by iizukanao** [comment]
node-rtsp-rtmp-server supports RTSP over HTTP as in the spec you mentioned. I think QuickTime Player has dropped support for RTSP some years ago, but alternatively you can view an RTSP-over-HTTP stream in VLC media player (Preferences -> Show All -> Input / Codecs -> Demuxers -> RTP/RTSP -> Check "Tunnel RTSP and RTP over HTTP").

**Comment by jandrieu** [comment]
After a long break, I'm thinking your answer to issue #9 clarifies that I *can't* use this module to access a remote RTSP then package & send it out via HTTP. That would be awesome, but if it can't, it can't.

I'd appreciate pointers on how to do it with gstreamer, however. 

I read the section on https://github.com/iizukanao/node-rtsp-rtmp-server#from-rtsp-client and tried to get that working with gstreamer (using rtspsrc), without luck. 

I could start the server fine. (Apparently)

I also tried using a video testsrc:
```

gst-launch-0.10 videotestsrc ! qtdemux name=demux ! \
    flvmux name=mux streamable=true ! queue ! \
    rtmpsink location='rtmp://localhost/live/STREAM_NAME' demux. ! \
    multiqueue name=mq ! h264parse ! mux. demux. ! mq. mq. ! aacparse ! mux.

WARNING: erroneous pipeline: could not link videotestsrc0 to demux
```
Similarly, I tried reducing the path to a minimal configuration, but failed.

I *was* able to verify that my rtsp works via gstreamer with this:
```
gst-launch-0.10 rtspsrc location=rtsp://192.168.250.229:5000/dltv1 ! rtph264depay ! ffdec_h264 ! autovideosink
```
That works nicely. But trying to make something similar based on the pipeline in the documentation fails:
```
gst-launch-0.10 rtspsrc location=rtsp://192.168.250.229:5000/dltv1 ! rtph264depay ! ffdec_h264 ! \
 qtdemux name=demux ! \
 flvmux name=mux streamable=true ! queue ! \
 rtmpsink location='rtmp://localhost/live/STREAM_NAME' demux. ! \
 multiqueue name=mq ! h264parse ! mux. demux. ! mq. mq. ! aacparse ! mux.

WARNING: erroneous pipeline: could not link ffdec_h2640 to demux
```
I was *also* not able to get the basic example for gstreamer working.  The gstreamer ran
```
gst-launch-0.10 filesrc location=/home/joe/Software/rtsp-http/node-rtsp-rtmp-server/car.mp4 ! qtdemux name=demux !     flvmux name=mux streamable=true ! queue !     rtmpsink location='rtmp://localhost/live/STREAM_NAME' demux. !     multiqueue name=mq ! h264parse ! mux. demux. ! mq. mq. ! aacparse ! mux.
Setting pipeline to PAUSED ...
Pipeline is PREROLLING ...
```
But when I tried to open it with VLC, it failed.  Looking at the server output, it couldn't find the stream:
```
2017-03-20 19:19:38.859 [rtmp:client=fQTxOL2e] connected
[rtmp: 1 sessions]
 fQTxOL2e: addr=127.0.0.1 port=53167
2017-03-20 19:19:38.861 [rtmp:handshake] warning: unknown message format, assuming format 1
2017-03-20 19:19:38.863 [rtmp:receive] unknown (not implemented) AMF command: _checkbw
2017-03-20 19:19:38.863 [rtmp:client=fQTxOL2e] requested stream live/STREAM_NAME
2017-03-20 19:19:38.863 [rtmp:client=fQTxOL2e] error: stream not found: live/STREAM_NAME
2017-03-20 19:19:38.865 [rtmp:client=fQTxOL2e] socket error: Error: read ECONNRESET
2017-03-20 19:19:38.865 [rtmp:client=fQTxOL2e] disconnected
[rtmp: 0 sessions]
```

Any suggestions would be appreciated.

**Comment by jandrieu** [comment]
Consider the open parts of my question merged into #70. I was able to get the RTSP input mostly working. See #84 for my question about avoiding transcoding.

In case it helps others, I ran into two problems with my node version. 

First, I had the wrong node version and the npm install didn't report anything. Or at least, I didn't notice it.

Second, after using NVM to change my node version, it took a while to realize that NVM installs to the $USER/.nvm directory and sets up environment variables accessible to that user. Running "sudo node server.js" not only changed the user to root, it doesn't keep the env vars.  So, I was *still* running the rtsp-rtmp-server on the old node version.

The obvious result was that I kept getting errors from v0.10.40 after I updated with nvm twice: once to v4.8.0 and once to v0.12.0. Interesting bug/limitation with NVM. Maybe that note will help someone having similar problems.
